plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1680,6 +1669,14 @@ fc::time_point producer_plugin_impl::calculate_block_deadline( const fc::time_po
    }
 }
 
+bool producer_plugin_impl::start_block_interrupted( const fc::time_point& deadline ) const {
+   if( _pending_block_mode == pending_block_mode::producing ) {
+      return deadline <= fc::time_point::now();
+   }
+   // if we can produce then honor deadline so production starts on time
+   return (!_producers.empty() && deadline <= fc::time_point::now()) || _received_block;
```

Comment:
Why `!_producers.empty()` here? Shouldn't we interrupt whenever the deadline is passed?

`_received_block` will trigger interruption in both `producing` and `speculating` modes?

Reply:
In speculative mode (HEAD mode) you just keep using the same block until a block is received. There is no reason to start a new block until another block is received. 

Yes, `_received_block` will trigger interruption in either case.